store/multiversion/store.go

```diff
@@ -243,10 +243,13 @@ func (s *Store) GetIterateset(index int) Iterateset {
 
 // CollectIteratorItems implements MultiVersionStore. It will return a memDB containing all of the keys present in the multiversion store within the iteration range prior to (exclusive of) the index.
 func (s *Store) CollectIteratorItems(index int) *db.MemDB {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+
 	sortedItems := db.NewMemDB()
 
 	// get all writeset keys prior to index
-	keys := s.GetAllWritesetKeys()
+	keys := s.txWritesetKeys
```

Comment:
this will acquire the lock for longer right? whereas none of the below logic actually invokes resources protected by the lock? does it make sense to do this way?

Reply:
meant to point to lines 246/247 where lock is acquired

Reply:
In Go, maps are always returned by reference, not by value.  This means returning a reference via GetAllWritesetKeys doesn't give us a copy, and doesn't do anything to protect the map.  The concurrent-access here is the `keys[i]` moment.